hpc-cluster-setup/v3/compute-post-install.sh

```diff
@@ -0,0 +1,182 @@
+wget https://developer.download.nvidia.com/compute/cuda/11.3.0/local_installers/cuda_11.3.0_465.19.01_linux.run
+chmod +x cuda_11.3.0_465.19.01_linux.run
+sudo ./cuda_11.3.0_465.19.01_linux.run --silent --override --toolkit --samples --no-opengl-libs
+
+echo 'Building nccl'
+cd "$INSTALL_ROOT"/packages || exit
+git clone https://github.com/NVIDIA/nccl.git || echo ignored
+cd nccl || exit
+git checkout tags/v2.11.4-1 -b v2.11.4-1
+# Choose compute capability 70 for Tesla V100 and 80 for Tesla A100
+# Refer https://en.wikipedia.org/wiki/CUDA#Supported_GPUs for different architecture
+make -j src.build NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_80,code=sm_80"
+make pkg.txz.build
+cd build/pkg/txz || exit
+
+tar xvfJ nccl_2.11.4-1+cuda11.3_x86_64.txz
+sudo cp -r nccl_2.11.4-1+cuda11.3_x86_64/include/* /usr/local/cuda/include/
+sudo cp -r nccl_2.11.4-1+cuda11.3_x86_64/lib/* /usr/local/cuda/lib64/
+
+echo 'Building aws-ofi-nccl'
+cd "$INSTALL_ROOT"/packages || exit
+git clone https://github.com/aws/aws-ofi-nccl.git || echo exists
+cd aws-ofi-nccl || exit
+git checkout aws
+git pull
+./autogen.sh
+
+./configure --prefix=/usr --with-mpi=/opt/amazon/openmpi --with-libfabric=/opt/amazon/efa/ --with-cuda=/usr/local/cuda --with-nccl=$INSTALL_ROOT/packages/nccl/build
+
+sudo yum install libudev-devel -y
+PATH=/opt/amazon/efa/bin:$PATH LDFLAGS="-L/opt/amazon/efa/lib64" make MPI=1 MPI_HOME=/opt/amazon/openmpi CUDA_HOME=/usr/local/cuda NCCL_HOME=$INSTALL_ROOT/packages/nccl/build
+sudo make install
+
+echo 'Installing bazel'
+sudo update-alternatives --set gcc "/usr/bin/gcc48"
+sudo update-alternatives --set g++ "/usr/bin/g++48"
+
+cd "$INSTALL_ROOT"/packages || exit
+echo 'downloading bazel'
+wget https://github.com/bazelbuild/bazel/releases/download/5.0.0/bazel-5.0.0-installer-linux-x86_64.sh
+sudo bash bazel-5.0.0-installer-linux-x86_64.sh
+
+sudo sh -c 'echo "/opt/amazon/openmpi/lib64/" > mpi.conf'
+sudo sh -c 'echo "$INSTALL_ROOT/packages/nccl/build/lib/" > nccl.conf'
+sudo sh -c 'echo "/usr/local/cuda/lib64/" > cuda.conf'
+sudo ldconfig
+
+cd /usr/local/lib || exit
+sudo rm -f ./libmpi.so
+sudo ln -s /opt/amazon/openmpi/lib64/libmpi.so ./libmpi.s
+
+
+echo 'installing NCCL'
+cd "$INSTALL_ROOT"/packages || exit
+git clone https://github.com/NVIDIA/nccl-tests.git || echo ignored
+cd nccl-tests || exit
+make MPI=1 MPI_HOME=/opt/amazon/openmpi CUDA_HOME=/usr/local/cuda NCCL_HOME="$INSTALL_ROOT"/packages/nccl/build
+
+echo "Installing additional packages"
+wget https://download-ib01.fedoraproject.org/pub/epel/7/x86_64/Packages/e/epel-release-7-14.noarch.rpm
+sudo rpm -Uvh epel-release*rpm
+sudo yum install nload -y
+
+sudo yum install -y mosh
+sudo yum install -y htop
+sudo yum install -y gdb
+sudo yum install -y tmux
+sudo yum install -y
+
+echo "================================"
+echo "===========Check EFA============"
+echo "================================"
+FI_EFA_USE_DEVICE_RDMA=1 fi_info -c FI_HMEM -p efa
```

Comment:
EFA related environment variables need to be set before this command can be run. Switch the order for setting on ENV variables